### PR TITLE
Test presence of repositories

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR adds an ability to test the presence of a repository, instead of finding it (which takes into account visibility of entity states).

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@dmdashenkov, PTAL.

File: server/src/main/java/io/spine/server/BoundedContext.java [hunk 1]
```diff
@@ -350,13 +352,42 @@ public CommandErrorHandler createCommandErrorHandler() {
...
+    /**
+     * Verifies if there a repository managing entities with the state of the passed class
+     * registered with this Bounded Context.
+     *
+     * <p>This method does not take into account visibility of entity states.
+     *
+     * @see #findRepository(Class)
+     */
+    @VisibleForTesting
+    public boolean hasRepositoryOfEntityWithState(Class<? extends Message> stateClass) {
```
Comment (dmdashenkov) [inline server/src/main/java/io/spine/server/BoundedContext.java]: This is too long. Also, the default behaviour for the `findRepository` method is to look up by the entity state type, thus the `has...` method looks up by the entity state type should have the shorter name.

File: server/src/test/java/io/spine/server/BoundedContextTest.java [hunk 2]
```diff
@@ -188,21 +186,38 @@ void projectionRepository() {
 
         <I, E extends Entity<I, ?>> void registerAndAssertRepository(Class<E> cls) {
             boundedContext.register(DefaultRepository.of(cls));
-            assertRegisteredRepositoryOf(cls);
+            assertThat(boundedContext.hasRepositoryOf(cls))
+                    .isTrue();
```
Comment (dmdashenkov) [inline server/src/test/java/io/spine/server/BoundedContextTest.java]: `assertTrue` is even better, I think. It's shorter and easier to read.

File: server/src/test/java/io/spine/server/BoundedContextTest.java [hunk 3]
```diff
@@ -324,63 +339,73 @@ void useEventStoreIfSet() {
...
-                                               .isMultitenant());
+            context = singleTenant();
 
-            bc = BoundedContext.newBuilder()
-                               .setMultitenant(false)
-                               .build();
+            assertMultitenancyEqual(context::isMultitenant, context.stand()::isMultitenant);
+        }
+
+        void assertMultitenancyEqual(Supplier<Boolean> s1, Supplier<Boolean> s2) {
```
Comment (dmdashenkov) [inline server/src/test/java/io/spine/server/BoundedContextTest.java]: Please use `BooleanSupplier` instead.

File: server/src/test/java/io/spine/server/BoundedContextTest.java [hunk 4]
```diff
@@ -324,63 +339,73 @@ void useEventStoreIfSet() {
...
+                    .build();
         }
+
+        private BoundedContext singleTenant() {
+            return BoundedContext
+                    .newBuilder()
+                    .setMultitenant(false)
+                    .build();
+        }
+
```
Comment (dmdashenkov) [inline server/src/test/java/io/spine/server/BoundedContextTest.java]: This line is redundant.

**Pull request merged**